src/posit/connect/oauth/oauth.py

```diff
@@ -4,26 +4,30 @@
 
 from typing_extensions import TypedDict
 
-from ..resources import ResourceParameters, Resources
+from .._types_context import ContextP
+from ..context import Context
 from .integrations import Integrations
 from .sessions import Sessions
 
 
-class OAuth(Resources):
-    def __init__(self, params: ResourceParameters, api_key: str) -> None:
-        super().__init__(params)
+class OAuth(ContextP[Context]):
+    def __init__(self, ctx: Context, api_key: str) -> None:
+        super().__init__()
+        self._ctx = ctx
+
+        # TODO-barret-q: Is this used?
         self.api_key = api_key
```

Comment:
I couldn't see where this is used. Just wanted to be sure I didn't lose this one as it is related to auth.